components/brave_wallet/browser/eth_allowance_manager.h

```diff
@@ -0,0 +1,122 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ETH_ALLOWANCE_MANAGER_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ETH_ALLOWANCE_MANAGER_H_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "base/barrier_callback.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/weak_ptr.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "brave/components/brave_wallet/common/brave_wallet_types.h"
+
+class PrefService;
+
+namespace brave_wallet {
+
+class JsonRpcService;
+class KeyringService;
+class BraveWalletService;
+
+class EthAllowanceManager {
+ public:
+  EthAllowanceManager(BraveWalletService* wallet_service,
```

Comment:
EthAllowanceManager does not need dependency on BraveWalletService, please remove